### ISSUE TITLE
Investigate need for Apple Silicon fix to `Internal_Convertfwrite(...)`

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/runtime#42954]
During AppleSilicon initial bring-up we observed a test failure where the `Internal_Convertfwrite(...)` function was being called with an empty string.  This was being treated as a failure during the MultiByte to utf-8 conversion.  For the conversion reporting a failure in this case was incorrect.  Therefore a fix was included in #40435.

However, it is not clear why this only occurred for `osx-arm64` and not `osx-x64`.

Temporarily remove the fix and investigate the root cause of why this only occurs on `arm64`.  We would like to be sure the fix is not masking another underlying issue.

See discussion https://github.com/dotnet/runtime/pull/40435/files/e186f0106acfd8ebba311581ddbc4e0877442f46#r498157083

**Comment by sdmaclea** [comment]
One of the tests which showed this issue was `readytorun/tests/mainv1/mainv1.sh`

```
{0x425774-0x10056cfcc} ASSERT [CRT    ] at /Users/steve/git/runtime/src/coreclr/src/pal/src/cruntime/printfcpp.cpp.70: 
WideCharToMultiByte failed.  Error is 42424
./mainv1.sh: line 140: 36999 Trace/BPT trap: 5       $CORE_ROOT/crossgen -readytorun -platform_assemblies_paths $CORE_ROOT:`pwd` -out test.ni.dll test.dll
```

/cc @janvorli

**Comment by sdmaclea** [comment]
There have been a lot of runtime instability issues where references were getting zeroed.  Most seem to be fixed by `macOS 11.3 Beta 6`.  I do not think this early failure in 11.0 Beta is worth investigating.

Closing.
